src/checker.ts

```diff
@@ -315,7 +315,14 @@ export class LicenseChecker extends EventEmitter {
     localDirectory: string | null,
     ...parents: string[]
   ): Promise<void> {
-    const pj: PackageJson = ensurePackageJson(json);
+    const pj: PackageJson =
+      packageName && this.isPackageWhitelisted(packageName)
+        ? {
+            name: packageName,
+            version: '0.0.0',
+            ...json,
+          }
+        : ensurePackageJson(json);
     if (!packageName) {
```

Comment:
To avoid the extra check above, could we pull this up to the start of the function and instead say:
```js
packageName = packageName || pj.name;
```

Reply:
pj is not defined at the start of the function. I may be missing the point there though.